src/main/java/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait.java

```diff
@@ -317,6 +319,111 @@ public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> orig
         }
     }
 
+    /**
+     * An {@link SCMHeadAuthority} that trusts contributors to the repository as well as those on a whitelist.
+     */
+    public static class TrustContributorsAndWhitelist extends TrustContributors {
+
+        private WhitelistSource whitelist;
+
+        /**
+         * Constructor.
+         */
+        @Inject
```

Comment:
do not use `@Inject`

Reply:
@stephenc Would you mind explaining the correct way to do general purpose dependency injection in Jenkins? I'm writing this logic into a new plugin yet DI in Jenkins is poorly documented.

Reply:
Don’t do dependency injection in Jenkins. Just do lookup by ExtensionList... much much safer